Fix a few test failures on big-endian systems

musicinmybrain opened the pull request:
These are all due to tests expecting little-endian dtypes, where in fact the endianness of the dtype is that of the host.

- [ ] closes #xxxx (Replace xxxx with the Github issue number) **No issue filed**
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature **Not applicable; fixes some tests on big-endian platforms**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

musicinmybrain committed:
Fix a few test failures on big-endian systems

These are all due to tests expecting little-endian dtypes, where in fact
the endianness of the dtype is that of the host.

jreback commented on pandas/tests/arrays/boolean/test_astype.py:
```diff
@@ -20,7 +22,8 @@ def test_astype():
 
     result = arr.astype("str")
-    expected = np.array(["True", "False", "<NA>"], dtype="<U5")
+    endian = {"little": "<", "big": ">"}[byteorder]
```
why don't we define this in https://github.com/pandas-dev/pandas/blob/main/pandas/_testing/__init__.py

e.g. make a module level ENDIAN and just use it here (and in other locations where you are doing similar things)

musicinmybrain replied:
Good plan. I’ll do that.

musicinmybrain committed:
Set ENDIAN in pandas._testing

Setting an ENDIAN constant here to represent the character for
host-endianness to use for numpy dtypes ("<" or ">") saves an import and
a little duplicated code across a number of test modules.

musicinmybrain commented:
The force push was to remove a stray `from sys import byteorder` left over from when I rebased this from `v1.3.5` to `main`.

jreback reviewed (approved):
lgtm thanks @musicinmybrain
